broker-daemon/block-order-worker/index.js

```diff
@@ -316,6 +312,37 @@ class BlockOrderWorker extends EventEmitter {
     }
   }
 
+  /**
+   * Move a block order to a completed state if all orders have been completed
+   * @param {String} blockOrderId
+   */
+  async completeBlockOrder (blockOrderId) {
+    this.logger.info('Attempting to put block order in a completed state', { id: blockOrderId })
+
+    // TODO: move status to its own sublevel so it can be updated atomically
+    try {
+      var value = await promisify(this.store.get)(blockOrderId)
+    } catch (e) {
+      if (e.notFound) {
+        this.logger.error('Attempted to move a block order to a completed state that does not exist', { id: blockOrderId })
+      } else {
+        this.logger.error('Error while retrieving block order to move it to a completed state', { id: blockOrderId, error: e.message })
+      }
+
+      throw e
+    }
+
+    const blockOrder = BlockOrder.fromStorage(blockOrderId, value)
+
+    // An order is only completed if all orders underneath the blockorder are out of
+    // an `ACTIVE` state.
+    blockOrder.complete()
```

Comment:
per our discussion, any validity checks should happen here (not in the model).

Also, we should check the actual amount of the block order and not just the underlying orders/fills.